include/no_os_eeprom.h

```diff
@@ -79,6 +81,8 @@ struct no_os_eeprom_desc {
 	const struct no_os_eeprom_platform_ops *platform_ops;
 	/** EEPROM extra parameters (device specific) */
 	void *extra;
+	/** EEPROM mutex*/
+	void 		*mutex;
```

Comment:
I would put the mutex as the first member of this structure to be able to lock the resource by a simple cast. It might be useful in the future.